gitignore-mode.el

```diff
@@ -48,7 +48,7 @@
   (set (make-local-variable 'conf-assignment-sign) nil))
 
 ;;;###autoload
-(dolist (pattern (list "/\\.gitignore\\'"
+(dolist (pattern (list "\\.gitignore.*\\'"
```

Comment:
[Please do not use one long line for your commit message body.]

> Currently the gitignore file extension regex detects if the
> extension starts with a slash, which of course no file extension
> extracted by emacs ever does.

The regexp in `auto-mode-alist` isn't matched against some concept of
an "extension"; it's matched against `buffer-file-name` (after some
minor processing).  By including the slash, the regexp is saying that
".gitignore" is the complete file name, sans its directory.

> In addition, this patch also recognize gitignore file conventions
> such as .gitignore_global, as created by SourceTree.

I think it'd be fine to add this as an entry, but I don't see the need
for the general wildcard regexp.